### ISSUE TITLE
Project descriptor generated by slurm cannot be parsed by BB

### ISSUE THREAD

**Issue by vulder** [PolyJIT/benchbuild#281]
The project identifier generated into the slurm script is not usable inside a bash script.
`$_project` is, for example, a ambiguous redirect. Further, even BB itself cannot correctly parse the identifier again*.

* this seems to be a problem with the escaping when calling BB inside the slurm bash script

Example identifier:
`'opus-c_projects@{'opus': f9d3d432d1}'`

Generated script example:
```
#!/bin/bash
#SBATCH --ntasks 1
#SBATCH --cpus-per-task 5
#SBATCH --array=0-0
#SBATCH --exclusive
#SBATCH --time=12:00:00
#SBATCH --nice=0

# List of available projects
projects=(
'opus-c_projects@{'opus': f9d3d432d1}'
)
# End of list of available projects
_project="${projects[$SLURM_ARRAY_TASK_ID]}"

exec 1> /path/username/vara-root/benchbuild/slurm-output/e5097ca4-16a0-40f4-9161-513a6a9d54da-$_project
exec 2>&1

...
```

Error:
```
slurmd/job200475/slurm_script: line 17: /path/username/vara-root/benchbuild/slurm-output/36e5fe2e-2887-4cd0-8d04-4690da06f315-$_project: ambiguous redirect
Fri 24 Jul 2020 04:38:52 PM CEST [kine06] prepare local node
kine06
Fri 24 Jul 2020 04:38:53 PM CEST [kine06] clean for kine06
Submitted batch job 200476
Traceback (most recent call last):
  File "/scratch/sattlerf/vara-virt/bin/benchbuild", line 8, in <module>
    sys.exit(main())
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/benchbuild/driver.py", line 9, in main
    return BenchBuild.run(*args)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/plumbum/cli/application.py", line 577, in run
    inst, retcode = subapp.run(argv, exit=False)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/plumbum/cli/application.py", line 572, in run
    retcode = inst.main(*tailargs)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/benchbuild/cli/run.py", line 76, in main
    prjs = project.populate(projects, group_names)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/benchbuild/project.py", line 483, in populate
    prj_cls = __add_filters__(prj_cls, version)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/benchbuild/project.py", line 432, in __add_filters__
    version_in = yaml.safe_load(version_str)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/composer.py", line 64, in compose_node
    if self.check_event(AliasEvent):
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/parser.py", line 574, in parse_flow_mapping_value
    return self.parse_flow_node()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/parser.py", line 268, in parse_flow_node
    return self.parse_node()
  File "/scratch/sattlerf/vara-virt/lib/python3.7/site-packages/yaml/parser.py", line 371, in parse_node
    token.start_mark)
yaml.parser.ParserError: while parsing a flow node
expected the node content, but found '<stream end>'
  in "<unicode string>", line 1, column 7:
    {opus:
          ^
srun: error: kine06: task 0: Exited with exit code 1
```

**Comment by simbuerg** [comment]
Should be fixed by #279

**Comment by vulder** [comment]
seems to work now :+1:
